vm-presets.md

```diff
@@ -0,0 +1,185 @@
+Virtual Machine Presets
+=============================
+
+`VirtualMachinePresets` are an extension to general `VirtualMachine`
+Configuration based heavily on `PodPresets` from Kubernetes. When a
+`VirtualMachine` is created, any applicable `VirtualMachinePresets`
+will be applied to the existing spec for the `VirtualMachine`. This allows
+for re-use of common settings that should apply to multiple `VirtualMachines`.
+
+
+Implementation
+------------------------
+
+`VirtualMachinePresets` are implemented as a Kubernetes `Initializer`. This
+means the `VirtualMachine` resource is modified before it is visible to or
+processed by any other component of KubeVirt.
+
+One noteworthy difference between the behavior of a `VirtualMachinePreset` and
+a `PodPreset` is that if an error occurs during processing of a `PodPreset`,
+Kubernetes will continue to create the `Pod` without applying the `PodPreset`.
+KubeVirt will move the VirtualMachine to the `Failed` state instead. The
+rationale behind this difference is that settings defined in a
+`VirtualMachinePreset` are presumed to be critical to the function of the
```

Comment:
I am not sure I can follow that rationale. What the VirtualMachinePreset applies, may or may not be critical for an Operating System inside the vm, but it should never be critical to launch a VirtualMachine.

Here one example flow where this is problematic:

 1. Create a vm
 2. Presets exand it
 3. I fetch the running vm spec
 4. I delete my vm
 5. Admin changes the presets
 6. I want to post the fetched spec again
 7. I can't because of a Preset merge conflict.

Reply:
Works for me.